### PR TITLE
Crystal field theory docs

### LINKED ISSUE #19498: Crystal Field documentation
More detailed documentation for the Crystal Field module has been requested - specifically including worked examples (in the `Interfaces` category along with the current documentation page). Also, some scientific background should be provided in a `Concepts` page.

### PR DESCRIPTION
**Pull request by mducle**
**Description of work.**

Adds a Crystal Field concepts page describing the theory behind the crystal field fit function calculations.

Also modified the links in the crystal fields techniques and interface page to reference each other.

**To test:**

Check that html documents (especially LaTeX / mathjax equations) build correctly.

Fixes #19498. 

*This does not require release notes* because it adds background documentation for already implemented functionality.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/concepts/CrystalField.rst [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+If you want to know how to make these calculations please see the :ref:`Python interface page <Crystal Field Python Interface>`
+or the :ref:`examples page <Crystal_Field_Examples>`.
+
+
+The Crystal Field Potential
+---------------------------
+
+The crystal field (or in older terminology, *crystalline electric field*)
+is the potential acting on a particular magnetic ion :math:`i` at position :math:`\mathbf{r}_i` in a solid or molecule.
+When first proposed by Van Vleck, it was thought to be an electrostatic effect due the charges
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:23]: A typo I think 'electrostatic effect due **to** the charges'

File: docs/source/concepts/CrystalField.rst [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+:math:`q_j` of neighbouring ions, located at :math:`\mathbf{r}_j`, of the magnetic ion (hence the original name).
+This leads to an approximation in which the surrounding ions are treated as point charges,
+so that the electric potential acting on the magnetic ion is
+
+.. math::
+    V_{\mathrm{CF}}(\mathbf{r}_i) = \frac{1}{4\pi\epsilon_0} \sum_j \frac{q_j}{|\mathbf{r}_j - \mathbf{r}_i|}.
+  :label: cfpot
+
+This potential satisfies Laplace's equation, :math:`\nabla^2 V(\mathbf{r}) = 0`,
+and so can expanded in terms of spherical harmonic functions
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:33]: 'and so can **be** expanded'

File: docs/source/concepts/CrystalField.rst [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+because the ligand electrons may be involved in bonding or charge transfer processes,
+and thus may have a spatial extent and may no longer be treated as point charges situated at the atomic sites.
+The formulism detailed above, however, was still found to be applicable to a wide variety of materials,
+*if* the radial part :math:`A_l^m r^l` is treated as a variable *crystal field parameter*.
+This is because the spherical harmonic functions form a complete orthogonal basis set,
+so that any function may be expressed as an expansion in them.
+Values obtained from equation :eq:`alm`, however, do not generally match experimentally determined crystal field parameters,
+except for some insulators and if the point charges :math:`q_j` are scaled from their nominal valences.
+
+Mantid contain routines which can calculate the :math:`A_l^m` parameters
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:56]: 'Mantid contain**s** routines'

File: docs/source/concepts/CrystalField.rst [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+
+At zero temperature only the ground state energy level is occupied by the electrons of the magnetic ion
+but as the temperature increases, higher energy excited levels will become occupied.
+The different levels have different magnetic moment expectation values,
+which leads to a change of the magnetic susceptibility of materials containing this magnetic ion with temperature
+as the population of the levels change.
+Because different magnetic materials have different crystal structures and hence a different local crystalline
+environment around the magnetic ions, they will also have different susceptibilities,
+which is determined in part by the crystal field.
+An applied magnetic field will change the energies of the levels with respects to each other (the Zeeman splitting)
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:76]: I think there should be no 's' at the end of 'with respects'

File: docs/source/concepts/CrystalField.rst [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+so the crystal field will also affect the magnetisation (the magnetic moment as a function of applied magnetic field).
+
+Inelastic neutron spectroscopy can measure the energy difference between these crystal field energy levels
+as neutrons may excited or de-excited electrons occupying one level to another.
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:80]: I think there is a typo in this sentence

File: docs/source/concepts/CrystalField.rst [hunk 6]
```diff
@@ -0,0 +1,603 @@
...
+  :label: bkq
+
+is often used as the crystal field parameter to be fitted, and this convention is used by Mantid
+(e.g. the fittable coefficients in Mantid are the :math:`B_q^k` rather than the :math:`A_q^k`).
+
+An alternative formulation of the crystal field Hamiltonian developed by Wybourne `[Wybourne65]`_, Judd `[Judd63]`_ and others used
+the spherical tensor operators :math:`T_q^{(k)}` instead of the Stevens operators.
+The matrix elements of the :math:`T_q^{(k)}` are then calculated directly using the Wigner-Eckart theorem.
+This is a faster calculation but results in different Hamiltonian matrix elements for a given set of crystal field parameter values.
+There are thus different "normalisation" of crystal field parameters depending on the formulism used to define the operators in the Hamiltonian.
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:160]: There is therefore a different "normalisation" of crystal field parameters ...

File: docs/source/concepts/CrystalField.rst [hunk 7]
```diff
@@ -0,0 +1,603 @@
...
+:math:`Y_{lm}` terms which are not invariant under the operations of the point group of the magnetic ion must thus be zero. 
+
+To determine the non-zero :math:`Y_{lm}` terms we need only consider the highest symmetry operation, since this is the most restrictive.
+Taking the principle axis as :math:`z`, the :math:`n`-fold rotations :math:`C_n` will change :math:`\phi` by :math:`2\pi/n`.
+Now, :math:`Y_{l,\pm m} \propto \exp(\pm i m \phi)`.
+Thus only terms with :math:`m` being an integer multiple of :math:`n` will be non-zero.
+A mirror plane perpendicular to :math:`x` will map :math:`x` to :math:`-x`, or :math:`\cos\phi\rightarrow-\cos\phi=\cos(\phi+\pi)`
+and so is equivalent to a rotation by :math:`180^\circ`.
+Thus any point group with a mirror plane must have only even :math:`m` terms,
+because we can always chose the crystal field coordinate system such that the mirror plane is the :math:`yz` plane
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:184]: choose instead of chose

File: docs/source/concepts/CrystalField.rst [hunk 8]
```diff
@@ -0,0 +1,603 @@
...
+(if there are no other symmetry in the system).
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:185]: symmetries instead of symmetry?

File: docs/source/concepts/CrystalField.rst [hunk 9]
```diff
@@ -0,0 +1,603 @@
...
+
+The roto-inversion operation :math:`S_n` is an :math:`n`-fold rotation followed by mirroring in the plane perpendicular to the rotation axis.
+This mirroring is actually equivalent to an inversion about the origin,
+mapping :math:`x` to :math:`-x`, :math:`y` to :math:`-y` and :math:`z` to :math:`-z`.
+In spherical coordinates this means :math:`\cos\theta\rightarrow-\cos\theta=` and :math:`\sin\theta\rightarrow-\sin\theta`.
+Thus the presence of a centre of inversion implies that only terms with even powers of :math:`\cos\theta` and :math:`\sin\theta`
+(e.g. terms with even :math:`l`) are allowed. 
+
+In fact, though, odd :math:`l` terms are always forbidden because the Stevens factor, :math:`\theta_l`, is zero for odd :math:`l` [#]_.
+This implies that, as far as the crystal field is concerned, all magnetic ions lies on centres of inversion.
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:195]: lie instead of lies?

File: docs/source/concepts/CrystalField.rst [hunk 10]
```diff
@@ -0,0 +1,603 @@
...
+along the axis of highest symmetry), rather than necessarily relating to any crystallographic axes. Thus it may be
+necessary to rotate the coordinate systems (or equivalently, the crystal field parameters) for actual calculations for
+magnetic fields applied parallel to particular crystallographic directions. 
+The `FOCUS manual <https://epubs.stfc.ac.uk/manifestation/5723/RAL-TR-95-023.pdf>`_ [#]_ 
+has some details of how these calculations may be accomplished, and further details may be found in reference `[Buckmaster72]`_.
+
+The :math:`\hat{J}_x`, :math:`\hat{J}_y`, :math:`\hat{J}_z` operators may be identified with the :math:`\hat{C}_q^{(k)}` operators for :math:`k=1`
+with :math:`x`, :math:`y` and :math:`z` corresponding to :math:`q=1,-1` and 0 respectively [#]_.
+
+The magnetisation then may be calculated from the expectation value of the magnetic moment operator :math:`\mathbf{J} = \mathbf{L} + 2\mathbf{S}`
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:234]: 'may then' instead of 'then may'

File: docs/source/concepts/CrystalField.rst [hunk 11]
```diff
@@ -0,0 +1,603 @@
...
+       where :math:`l` is the orbital quantum number of the single electron (e.g. 3 for :math:`f` electrons).
+       The :math:`3j` symbol in this expression is zero unless :math:`k` is even.
+       Moreover, the top row of the :math:`3j` symbol must obey the *triangular inequality* which in this case implies that :math:`k<2l`.
+       Thus for rare earths, only the terms :math:`k=2,4,6` are needed.
+       The :math:`k=0` term is a constant and does not produce any splitting, rather it shifts the energy of all levels by a constant,
+       so is usually ignored in neutron spectroscopy (which can only measure the difference between energy levels).
+       Note also that the above term is not calculated in Mantid or for the splitting of the ground state multiplet.
+       This is because it only contributes to the Stevens factor which, as per equation :eq:`bkq`,
+       is usually absorbed into the crystal field parameter :math:`B_q^k`.
+       Thus, equation :eq:`redmat1e` implies that, as far as the crystal field is concerned, all magnetic ions lies on centres of inversion.
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:304]: lie instead of lies?

File: docs/source/concepts/CrystalField.rst [hunk 12]
```diff
@@ -0,0 +1,603 @@
...
+
+.. [#] The Mantid crystal field code is a port to C++ of the FOCUS Fortran 77 code by Peter Fabi.
+
+.. [#] Note that equation :eq:`redmat1e` applies only for the orbital part. In this case we apply the rank 1 operator mostly to the spin part where
+       the reduced matrix element is non-zero, and this is subsumed into the total angular momentum :math:`J`
+
+Appendix A: Wybourne Normalisation
+----------------------------------
+
+It turns out that the spherical harmonic functions :math:`Y_{lm}` is not the most convenient form in which to express the
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:314]: are instead of is?

File: docs/source/concepts/CrystalField.rst [hunk 13]
```diff
@@ -0,0 +1,603 @@
...
+Note that the operators :math:`\hat{C}_q^{(k)}` are *not* the Stevens operators :math:`O_q^(k)`.
+This is because although Stevens constructed his operators from the tesseral harmonics functions,
+he omits the prefactors of those equations using only the parts containing the :math:`x`, :math:`y`, :math:`z` coordinates.
+The :math:`\hat{C}_q^{(k)}` *does* contain the prefactors, so are related to the Stevens operators by:
+
+.. math::
+    O_q^{(k)} = \lambda_{k,|q|} \hat{C}_q^{(k)}
+  :label: wy2stev
+
+where the ratio :math:`\lambda_{k,|q|}` are summarised in `Table 3`_.
```
Comment (robertapplin) [inline docs/source/concepts/CrystalField.rst:399]: is instead of are?

**Review by robertapplin — CHANGES_REQUESTED** [review]
Nice addition to the crystal field documentation. Just a few comments on possible typos

File: docs/source/concepts/CrystalField.rst (hunk 5 above)
Comment (mducle) [reply docs/source/concepts/CrystalField.rst:80]: yes... thanks for spotting!

**Comment by mducle** [comment]
@robertapplin Thanks for the close reading! I've now read through it again and made some more corrections - mostly to the equations...

**Review by robertapplin — APPROVED** [review]
Thanks, looks good. The equations are formatted correctly and all the links are working 👍

**Pull request merged**